lib/hutch/tracers/opbeat.rb

```diff
@@ -0,0 +1,39 @@
+require 'opbeat'
+
+module Hutch
+  module Tracers
+    # Tracer for Opbeat, which traces each message processed.
+    class Opbeat
+      KIND = 'messaging.hutch'.freeze
+
+      # @param klass [Consumer] Consumer instance (!)
+      def initialize(klass)
+        @klass = klass
+      end
+
+      # @param message [Message]
+      def handle(message)
+        ::Opbeat.transaction(sig, KIND, extra: extra_from(message)) do
+          ::Opbeat.trace(sig, KIND, extra: extra_from(message)) do
+            @klass.process(message)
+          end
+        end.done(true)
+      end
+
+      private
+
+      def sig
+        @klass.class.name
+      end
+
+      def extra_from(message)
+        {
+          body: message.body.to_s,
```

Comment:
Scrubbable how? You don't need the trace, actually. Transaction automatically have one root trace, so adding another one is probably redundant. I can see how I could've let you to add one. Sorry.